### DISCUSSION TITLE
Issue building via platformIO `0.6.3` lanbon_l8

### DISCUSSION THREAD

**Discussion by samturner3** [HASwitchPlate/openHASP#350] [Q&A] [answered]
On branch 0.6.3
Using `[env:lanbon_l8]` build
![Screen Shot 2022-07-07 at 7 28 20 pm](https://user-images.githubusercontent.com/14887388/177741063-1716c4b8-25da-4dde-b01e-4583098ffa26.png)
Results in error
```
#include "lv_fs_if.h"
          ^~~~~~~~~~~~
compilation terminated.
*** [.pio/build/lanbon_l8/lib804/freetype/base/ftinit.c.o] Error 1
In file included from lib/freetype/include/freetype/config/ftconfig.h:42,
                 from lib/freetype/src/base/ftsystem.c:29:
lib/freetype/devel/ftstdlib.h:37:10: fatal error: lv_fs_if.h: No such file or directory
```
Only change I made was `include/user_config_override.h` and `platformio_override.ini` with ` -DUSE_CONFIG_OVERRIDE` uncommented. 

I have run `git submodule update --init --recursive` with no change.

**Comment by samturner3** [comment, marked as answer]
It does build however on `0.6.2` with warnings about 
```
warning: #warning >>>>------>> TOUCH_CS pin not defined, TFT_eSPI touch functions will not be available! [-Wcpp]
       #warning >>>>------>> TOUCH_CS pin not defined, TFT_eSPI touch functions will not be available!
```
(Not sure if relevant)        

So looks like an issue with `0.6.3`. Is there another build that would work on the lanbon l8 ?

**Reply by fvanroie** [reply]
It seems you are missing the FreeType submodule. Did you [clone openHASP](https://www.openhasp.com/0.6.3/compiling/local/) with the `--recursive` option?

If you already cloned openHASP without the <code>--recursive</code> parameter, you can fetch the submodules separately using:
```sh
git submodule update --init --recursive
```

**Reply by samturner3** [reply]
I have done above yes

**Reply by samturner3** [reply]
It worked on my M1 mac so all good.

For anyone else my next troubleshooting step would be to remove the repo and clone from fresh (not forgetting to use the `--recursive` flag of course)

**Reply by bensuffolk** [reply]
I have just tried to compile the latest revision, and am having the same problem on my intel mac. I have tried a "clean all" and the submodule update has been done.

```
In file included from lib/freetype/include/freetype/config/ftconfig.h:42,
                 from lib/freetype/include/freetype/internal/ftdebug.h:29,
                 from lib/freetype/src/base/ftadvanc.c:19,
                 from lib/freetype/src/base/ftbase.c:21:
lib/freetype/devel/ftstdlib.h:37:10: fatal error: lv_fs_if.h: No such file or directory
```

The missing file is there though:

```
$ find . -name "lv_fs_if.h"
./lib/lv_fs_if/lv_fs_if.h
```

Removing the repo and cloning again from scratch (including the --recursive) as suggested by @samturner3 yields the same results.

**Reply by fvanroie** [reply]
Something broke, not sure where yet... Even rebuilding older commits that worked before, now fail. This leads me to think there's some update from VScode or PIO that changes the include paths or something. Another possibility is an external library that makes the build go haywire.

**Reply by bensuffolk** [reply]
That makes sense both vs code and pio have done self updates recently on my machine.

I spent an hour or so trying to find fault and looking at include paths, but no joy yet. I hope you have better luck!

**Reply by samturner3** [reply]
ok so maybe my M1 mac just haden't pulled those updates yet so may not be an intel vs m1 thing.

**Reply by fvanroie** [reply]
It's due to the way Freetype links into several libraries needed to build the project.
Something changed that it can't find all the dependencies anymore, creating an *include hell*.
I've disabled the building of Freetype for the time being, so openHASP is able to build successfully.

**Reply by bensuffolk** [reply]
I see that we have the build flag in _esp32.ini
```
-D HASP_USE_FREETYPE=0
```

But for me I'm still getting the error. 

The dependency tree still shows it :
```
Dependency Graph
|-- ArduinoJson @ 6.19.4
|-- lvgl @ 7.11.1+sha.c539546
|-- lv_lib_png @ 0.0.0+20220724082124.sha.d42282b
|   |-- lvgl @ 7.11.1+sha.c539546
|-- lv_lib_bmp @ 0.0.0+20220724082125.sha.b6ca85d
|   |-- lvgl @ 7.11.1+sha.c539546
|-- lv_lib_gif @ 0.0.0+20220724082126.sha.afdc2d6
|   |-- lvgl @ 7.11.1+sha.c539546
|-- ConsoleInput @ 0.0.3+sha.79180a9
|-- AceButton @ 1.9.2
|-- StreamUtils @ 1.6.3
|   |-- EEPROM @ 2.0.0
|-- PubSubClient @ 2.8.0
|-- LovyanGFX @ 0.4.18
|   |-- SPI @ 2.0.0
|   |-- Wire @ 2.0.0
|-- FT6336U CTP Controller @ 1.0.2+sha.d96d4be
|   |-- Wire @ 2.0.0
|-- WiFi @ 2.0.0
|-- ArduinoLog
|-- Wire @ 2.0.0
|-- SPI @ 2.0.0
|-- Preferences @ 2.0.0
|-- EEPROM @ 2.0.0
|-- lv_fs_if @ 0.0.0+20220202035348
|   |-- lvgl @ 7.11.1+sha.c539546
|-- HTTPClient @ 2.0.0
|   |-- WiFi @ 2.0.0
|   |-- WiFiClientSecure @ 2.0.0
|   |   |-- WiFi @ 2.0.0
|-- CharStream
|-- StringStream
|-- freetype @ 2.10.4
|-- lv_lib_freetype
|   |-- lvgl @ 7.11.1+sha.c539546
|   |-- freetype @ 2.10.4
|   |-- lv_fs_if @ 0.0.0+20220202035348
|   |   |-- lvgl @ 7.11.1+sha.c539546
|-- lv_lib_qrcode
|   |-- lvgl @ 7.11.1+sha.c539546
|-- WiFiClientSecure @ 2.0.0
|   |-- WiFi @ 2.0.0
|-- SimpleFTPServer
|   |-- FFat @ 2.0.0
|   |   |-- FS @ 2.0.0
|   |-- FS @ 2.0.0
|   |-- LittleFS @ 2.0.0
|   |   |-- FS @ 2.0.0
|   |-- Ethernet @ 2.0.0
|   |   |-- WiFi @ 2.0.0
|   |-- SD @ 2.0.0
|   |   |-- FS @ 2.0.0
|   |   |-- SPI @ 2.0.0
|   |-- SPI @ 2.0.0
|   |-- SPIFFS @ 2.0.0
|   |   |-- FS @ 2.0.0
|   |-- WiFi @ 2.0.0
|-- Update @ 2.0.0
|-- DNSServer @ 2.0.0
|   |-- WiFi @ 2.0.0
|-- WebServer @ 2.0.0
|   |-- WiFi @ 2.0.0
|   |-- FS @ 2.0.0
|-- ESPmDNS @ 2.0.0
|   |-- WiFi @ 2.0.0
|-- ArduinoOTA @ 2.0.0
|   |-- Update @ 2.0.0
|   |-- WiFi @ 2.0.0
|   |-- ESPmDNS @ 2.0.0
|   |   |-- WiFi @ 2.0.0
|-- HTTPUpdate @ 2.0.0
|   |-- HTTPClient @ 2.0.0
|   |   |-- WiFi @ 2.0.0
|   |   |-- WiFiClientSecure @ 2.0.0
|   |   |   |-- WiFi @ 2.0.0
|   |-- Update @ 2.0.0
|   |-- WiFi @ 2.0.0
|-- Ethernet @ 2.0.0
|   |-- WiFi @ 2.0.0
|-- ETHSPI
|   |-- WiFi @ 2.0.0
|-- SPIFFS @ 2.0.0
|   |-- FS @ 2.0.0
|-- FS @ 2.0.0
|-- LittleFS @ 2.0.0
|   |-- FS @ 2.0.0
```

Is there something else I need to do to disable the building / inclusion of FreeType?

**Reply by bensuffolk** [reply]
I have delete freetype from the lib directory and that seems to have done the trick.
